src/main/java/org/graylog2/gelfclient/transport/GelfSenderThread.java

```diff
@@ -139,4 +146,40 @@ public void stop() {
         keepRunning.set(false);
         senderThread.interrupt();
     }
+
+    /**
+     * Block and wait for all messages in the queue to send. This can be used during shutdown to wait for
+     * queued messages to send before shutting down.
+     * Waiting will continue for the indicated {@code FLUSH_WAIT_RETRIES} and {@code FLUSH_WAIT_MS.}
+     */
+    void flushSynchronously() {
+
+        for (int i = 0; i <= FLUSH_WAIT_RETRIES; i++) {
+            if (!flushingInProgress()) {
+                LOG.debug("Successfully flushed messages. Shutting down now.");
+                continue;
+            }
+
+            LOG.debug("Flushing in progress. [{}] messages are still enqueued, and [{}] messages are still in-flight.",
+                      queue.size(), inflightSends.get());
+
+            try {
+                Thread.sleep(FLUSH_WAIT_MS);
+            } catch (InterruptedException e) {
+                LOG.error("Interrupted message flushing during shutdown after [{}}] attempts.", i);
+            }
+
+            if (i == FLUSH_WAIT_MS) {
```

Comment:
I guess this should be `FLUSH_WAIT_RETRIES`?
```suggestion
            if (i == FLUSH_WAIT_RETRIES) {
```